pints/tests/test_nested_ellipsoid_sampler.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+#
+# Tests ellipsoidal nested sampler.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import unittest
+import numpy as np
+
+import pints
+import pints.toy
+
+# Unit testing in Python 2 and 3
+try:
+    unittest.TestCase.assertRaisesRegex
+except AttributeError:
+    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
+
+debug = False
+
+
+class TestNestedEllipsoidSampler(unittest.TestCase):
+    """
+    Unit (not functional!) tests for :class:`NestedEllipsoidSampler`.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """ Prepare for the test. """
+        # Create toy model
+        model = pints.toy.LogisticModel()
+        cls.real_parameters = [0.015, 500]
+        times = np.linspace(0, 1000, 1000)
+        values = model.simulate(cls.real_parameters, times)
+
+        # Add noise
+        np.random.seed(1)
+        cls.noise = 10
+        values += np.random.normal(0, cls.noise, values.shape)
+        cls.real_parameters.append(cls.noise)
+
+        # Create an object with links to the model and time series
+        problem = pints.SingleOutputProblem(model, times, values)
+
+        # Create a uniform prior over both the parameters and the new noise
+        # variable
+        cls.log_prior = pints.UniformLogPrior(
+            [0.01, 400],
+            [0.02, 600]
+        )
+
+        # Create a log-likelihood
+        cls.log_likelihood = pints.GaussianKnownSigmaLogLikelihood(
+            problem, cls.noise)
+
+    def test_construction_errors(self):
+        # Tests if invalid constructor calls are picked up.
+
+        # First arg must be a log likelihood
+        self.assertRaisesRegex(
+            ValueError, 'must extend pints.LogPrior',
+            pints.NestedEllipsoidSampler, self.log_likelihood)
+
+    def test_hyper_params(self):
+        # Tests the hyper parameter interface is working.
+        sampler = pints.NestedEllipsoidSampler(self.log_prior)
+        self.assertEqual(sampler.n_hyper_parameters(), 6)
+        sampler.set_hyper_parameters([220, 130, 2.0, 133, 1, 0.8])
+        self.assertEqual(sampler.n_active_points(), 220)
+        self.assertEqual(sampler.n_rejection_samples(), 130)
+        self.assertEqual(sampler.enlargement_factor(), 2.0)
+        self.assertEqual(sampler.ellipsoid_update_gap(), 133)
+        self.assertTrue(sampler.dynamic_enlargement_factor())
+        self.assertTrue(sampler.alpha(), 0.8)
+
+    def test_getters_and_setters(self):
```

Comment:
Please break up into a separate method per getter/setter pair

Reply:
Really? We do this all over the place at the moment. Will end up making the test files quite a bit longer. At the moment, it's easy enough to determine what causes a test to fail by finding the line number. Happy for you to do this though if you feel strongly.